cl_dll/discord_integration.cpp

```diff
@@ -343,17 +349,26 @@ namespace discord_integration
 					get_map_name(map_name, ARRAYSIZE(map_name));
 					if (map_name[0])
 					{
-						// adjust to lowercase
-						unsigned char *tptr = (unsigned char *)map_name;
-						while (*tptr) {
-							*tptr = tolower(*tptr);
-							tptr++;
-						}
+						// We specifically don't want to convert the map name that will be shown when hovering over the map icon (presence.largeImageText), so that why it got moved above.
+						presence.largeImageText = map_name;
+
+						gHUD.ConvertToLowerCase(map_name);
 
 						if (maps_with_thumbnails.find(map_name) != maps_with_thumbnails.cend())
+						{
 							presence.largeImageKey = map_name;
-
-						presence.largeImageText = map_name;
+						}
+						else
+						{
+							auto closest_match = std::find_if(map_series_with_thumbnails.begin(), map_series_with_thumbnails.end(), [&](const std::string& str) {
+    												return static_cast<std::string>(map_name).find(str) != std::string::npos; // hl1_bhop_am_beta6 -> hl1_bhop_am
```

Comment:
This finds not just the prefix, but any substring, right? Is that intended?

Reply:
> This finds not just the prefix, but any substring, right?

That's not true? Here's an example that I made: https://godbolt.org/z/7Yh6xsWqv

But if you don't like this option, you can of course use one of the next followings to implement that too:

- std::mismatch
- std::equal
- .compare

Reply:
It finds with prefix too:

![image](https://github.com/YaLTeR/OpenAG/assets/1794388/21f542f9-e8f5-407e-b6a8-1289d6b4f549)

But we only want with suffix, right?

Reply:
Damn, how I did not thinked about such case at first, yeah then you are right, I would to fix it with one of options that I mentioned above then